### DISCUSSION TITLE
Avoid routing on loader error

### DISCUSSION THREAD

**Discussion by Nullpo** [remix-run/remix#8780] [Q&A]
I have a view called 'A' and a view called 'B'.

The 'A' view navigates to the view 'B'

But the view 'B' had a error getting the data IN THE LOADER

So I want to know how can I avoid to change the page. Not ony the URL but I need to mantain the state of the 'A' view

**Comment by Erij-Maherzia-BEN-BRAHIM** [comment]
before making the redirection, you can fetch the data that you will fetch on the route B on the route A
if the response is ok make the redirection else return null
